bundle-core/src/main/java/net/discdd/utils/BundleUtils.java

```diff
@@ -327,15 +332,15 @@ public static void createBundlePayloadForAdus(List<ADU> adus, byte[] routingData
     public static void encryptPayloadAndCreateBundle(Encrypter payloadEncryptor, ECPublicKey clientIdentityPublicKey,
                                                      ECPublicKey clientBaseKeyPairPublicKey,
                                                      ECPublicKey serverIdentityPublicKey, String encryptedBundleId,
-                                                     byte[] payloadBytes, OutputStream outputStream) throws IOException, NoSuchAlgorithmException, InvalidKeyException {
+                                                     InputStream payloadStream, OutputStream outputStream) throws IOException, NoSuchAlgorithmException, InvalidKeyException, InvalidMessageException, LegacyMessageException {
```

Comment:
let's take a look at LegacyMessageException. it would be nice to avoid adding a new exception